File: javatest/src/main/java/com/anymore/javatest/di/module/ViewModelFactoryModule.java
package com.anymore.javatest.di.module;

import android.arch.lifecycle.ViewModelProvider;
import com.anymore.javatest.ViewModelFactory;
import dagger.Binds;
import dagger.Module;

/**
 * Created by liuyuanmao on 2019/2/22.
 */
@Module
public abstract class ViewModelFactoryModule {

    @Binds
    public abstract ViewModelProvider.Factory provideViewModelFactory(ViewModelFactory factory);
}

File: mvvmkit/src/main/java/com/anymore/mvvmkit/di/module/ViewModelFactoryModule.kt
package com.anymore.mvvmkit.di.module

import android.arch.lifecycle.ViewModelProvider
import com.anymore.mvvmkit.mvvm.ViewModelFactory
import dagger.Binds
import dagger.Module

/**
 * Created by liuyuanmao on 2019/2/21.
 */
@Module
interface ViewModelFactoryModule {
    @Binds
    fun provideViewModelFactory(viewModelFactory: ViewModelFactory):ViewModelProvider.Factory
}